File: mainscreen/src/main/java/packi/day/main/MainActivity.java
package packi.day.main;

import android.os.Bundle;
import android.support.v4.app.FragmentTransaction;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;

import packi.day.store.DataStore;
import packi.day.store.feature.HashMapStoreDelegate;

public class MainActivity extends AppCompatActivity implements StoreLocator {

    private DataStore store;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        store = new DataStore(getApplicationContext(), new HashMapStoreDelegate());

        Toolbar viewById = findViewById(R.id.toolbar);
        viewById.setTitle(R.string.app_name);

        FragmentTransaction ft = getSupportFragmentManager().beginTransaction();
        ft.replace(R.id.content, new FocusCelebrationFragment());
        ft.commit();
    }

    @Override
    public DataStore getStore() {
        return store;
    }

}

File: mainscreen/src/main/java/packi/day/main/MainActivity.kt
package packi.day.main

import android.annotation.SuppressLint
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.Toolbar
import com.google.android.gms.analytics.GoogleAnalytics
import packi.day.common.report

import packi.day.store.InternationalDayRepository
import packi.day.store.StoreLocator

/// Class is actually registered
@SuppressLint("Registered")
class MainActivity : AppCompatActivity(), StoreLocator {

    override lateinit var store: InternationalDayRepository

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        store = InternationalDayRepository(applicationContext)

        val viewById = findViewById<Toolbar>(R.id.toolbar)
        viewById.setTitle(R.string.app_name)

        supportFragmentManager.beginTransaction()
                .replace(R.id.content, FocusCelebrationView())
                .commit()
    }

    override fun onResume() {
        super.onResume()
        GoogleAnalytics.getInstance(applicationContext).report("/instantApp")
    }
}
